src/VisualStudio/CSharp/Impl/CodeModel/CSharpCodeModelService.CodeModelEventCollector.cs

```diff
@@ -893,26 +893,20 @@ protected override void EnqueueAddEvent(SyntaxNode node, SyntaxNode parent, Code
                 {
                     foreach (var attribute in ((AttributeListSyntax)node).Attributes)
                     {
-                        if (parent is BaseFieldDeclarationSyntax)
-                        {
-                            foreach (var variableDeclarator in ((BaseFieldDeclarationSyntax)parent).Declaration.Variables)
-                            {
-                                eventQueue.EnqueueAddEvent(attribute, variableDeclarator);
-                            }
-                        }
-                        else
-                        {
-                            eventQueue.EnqueueAddEvent(attribute, parent);
-                        }
+                        AddOrRemoveEventToEventQueueForAttributes(attribute, parent, eventQueue.EnqueueAddEvent);
                     }
                 }
+                else if (node is AttributeSyntax)
+                {
+                    AddOrRemoveEventToEventQueueForAttributes((AttributeSyntax)node, parent, eventQueue.EnqueueAddEvent);
+                }
                 else
                 {
                     eventQueue.EnqueueAddEvent(node, parent);
                 }
             }
 
-            protected override void EnqueueChangeEvent(SyntaxNode node, SyntaxNode parent, CodeModelEventType eventType, CodeModelEventQueue eventQueue)
+            protected override void EnqueueRemoveEvent(SyntaxNode node, SyntaxNode parent, CodeModelEventQueue eventQueue)
```

Comment:
This diff is confusing because you appear to have re-ordered `EnqueueChangeEvent` and `EnqueueRemoveEvent`.  Can you put them back in the original order?